postproengine/instantaneousplanes.py

```diff
@@ -186,24 +257,149 @@ def __init__(self, parent, inputs):
             self.iters    = self.parent.iters
             return
 
-    # This is not needed at the moment, keeping here just in case
-    #@registeraction(actionlist)
-    class action1():
-        actionname = 'action1'
-        blurb      = 'A description of action'
-        required   = True
+
+    @registeraction(actionlist)
+    class animate():
+        actionname = 'animate'
+        blurb      = 'Generate animation of range of cross flow planes'
+        required   = False
         actiondefs = [
-            {'key':'name',     'required':True,  'help':'An arbitrary name',  'default':''},
+            {'key':'name', 'required':True,  'help':'Name of video', 'default':'output.mp4'},
+            {'key':'fps', 'required':False,  'help':'Frame per second', 'default':1},
+            {'key':'imagefilename', 'required':True,  'help':'savefile name of images', 'default':''},
+            {'key':'times', 'required':False,  'help':'Override parent times for animation', 'default':None},
         ]
-        
         def __init__(self, parent, inputs):
             self.actiondict = mergedicts(inputs, self.actiondefs)
             self.parent = parent
             print('Initialized '+self.actionname+' inside '+parent.name)
-            print(self.actiondict)
             return
 
         def execute(self):
             print('Executing ' + self.actionname)
+            video_name = self.actiondict['name']
+            fps = self.actiondict['fps']
+            #filetype = self.actiondict['filetype']
+            imagefilename = self.actiondict['imagefilename']
```

Comment:
one convenience feature we can add is to have this action automatically create any necessary directories to store the image files beforehand.  Mostly because some lazy people like me don't always remember to create a subdirectory to store the images before running the scripts.

EDIT: this comment should actually be applied to the "plot" action above, where it's generating the images.

Reply:
Good suggestion! I've updated all three actions to automatically create the directory before saving if it does not already exist